backend/pkg/commons/db/statistics.go

```diff
@@ -841,34 +852,42 @@ func gatherValidatorDepositWithdrawals(day uint64, data []*types.ValidatorStatsT
 	depositsQry := `
 		with 
 		first_valid_deposits as (
-			select distinct on (publickey)
-				publickey,
-				block_slot,
-				block_index
+			select distinct on (blocks_deposits.publickey)
+				blocks_deposits.publickey,
+				blocks_deposits.block_slot,
+				blocks_deposits.block_index
 			from
 				blocks_deposits
+			inner join blocks on
+				blocks_deposits.block_root = blocks.blockroot and (blocks.status = '1' or blocks.slot = 0)
 			where
-				valid_signature
-				and publickey in (select bd.publickey from blocks_deposits bd where bd.valid_signature and bd.block_slot >= $1 and bd.block_slot <= $2)
```

Comment:
note: valid_signature check here is not needed since this query is only to get performance-gains (like noted in the comment in the patch)